op-deployer/pkg/deployer/flags.go

```diff
@@ -52,15 +53,14 @@ func NewDeploymentTarget(s string) (DeploymentTarget, error) {
 	}
 }
 
-var DefaultCacheDir string
-
-func init() {
-	var err error
+func GetDefaultCacheDir() string {
 	homeDir, err := os.UserHomeDir()
 	if err != nil {
-		panic(fmt.Sprintf("failed to get home directory: %s", err))
+		fallbackDir := ".op-deployer/cache"
+		log.Printf("error getting user home directory: %v, using fallback directory: %s\n", err, fallbackDir)
+		return fallbackDir
 	}
-	DefaultCacheDir = path.Join(homeDir, ".op-deployer/cache")
+	return path.Join(homeDir, ".op-deployer/cache")
```

Comment:
nit: since you're using `".op-deployer/cache"` multiple times, it'd probably be best to define it as a variable.